DOC: wording in linear model interpretation

GaelVaroquaux opened the pull request:
Minor changes in wording in the example. Nothing but English and a few statistical pointers.

GaelVaroquaux committed:
DOC: wording in linear model interpretation

glemaitre commented on examples/inspection/plot_linear_model_coefficient_interpretation.py:
```diff
@@ -92,21 +93,23 @@
-# to simplify our problem and approximate a normal distribution.
+# Looking closely at the WAGE distribution reveals that it has a
+# long tail. For this reason, we should take its logarithm
+# to turn it approximately into a normal distribution.
```
Do you think it could nice to add why a normal distribution is better (I assume this because this is an assumption of the linear model, normality of the residuals?)

glemaitre reviewed (approved):
I am happy with the changes.

glemaitre commented on examples/inspection/plot_linear_model_coefficient_interpretation.py:
```diff
@@ -296,8 +302,11 @@
-# We can check the coefficient variability through cross-validation.
-# If coefficients vary in a significant way changing the input dataset
+# We can check the coefficient variability through cross-validation:
+# it is a form of data perturbation (related to 
```
```suggestion
# it is a form of data perturbation (related to
```

I think this the issue with the linting

cmarmo commented on examples/inspection/plot_linear_model_coefficient_interpretation.py:
```diff
@@ -533,14 +544,17 @@
+# stable, as visible from data perturbations, in a
 # cross validation (see the :ref:`ridge_regression` User Guide section),
-# as is shown in the plot below to be compared with the
+# shown in the plot below and to be compared with the
```
This sentence is a bit obscure to me... probably I have to blame my English...

cmarmo commented on examples/inspection/plot_linear_model_coefficient_interpretation.py:
```diff
@@ -632,14 +646,25 @@
 # AGE and EXPERIENCE and suppresses one of them for the sake of the prediction.
 #
+# It is important to keep in mind that the coefficients that have been
+# dropped may still be related to the outcome in by themselves: the model
```
```suggestion
# dropped may still be related to the outcome by themselves: the model
```
not sure... just sound better to me

cmarmo commented on examples/inspection/plot_linear_model_coefficient_interpretation.py:
```diff
@@ -632,14 +646,25 @@
 # * Different linear models respond differently to feature correlation and
 #   coefficients could significantly vary from one another.
+# * Inspecting coefficients across the folds of a cross-validation loop
+#   gives an idea of their stability.
```
A question about this and all other recurrences of "stability" and derived words about coefficients.
Using it worthed me a [comment](https://github.com/scikit-learn/scikit-learn/pull/15706#discussion_r353219620), so I tried to avoid it... and now it's back... :) is there really a "truth" to look for? Thanks!

NicolasHug replied:
I'd say that's fine here

I suggested variance / variability because that's really what it's about, and it's important for section titles to be accurate. But stability in this context isn't ambiguous.

GaelVaroquaux committed:
Update examples/inspection/plot_linear_model_coefficient_interpretation.py

Co-Authored-By: Guillaume Lemaitre <g.lemaitre58@gmail.com>

GaelVaroquaux committed:
Update examples/inspection/plot_linear_model_coefficient_interpretation.py

Co-Authored-By: Guillaume Lemaitre <g.lemaitre58@gmail.com>

GaelVaroquaux committed:
Update examples/inspection/plot_linear_model_coefficient_interpretation.py

Co-Authored-By: Chiara Marmo <cmarmo@users.noreply.github.com>

GaelVaroquaux committed:
Update examples/inspection/plot_linear_model_coefficient_interpretation.py

Co-Authored-By: Chiara Marmo <cmarmo@users.noreply.github.com>

GaelVaroquaux committed:
DOC: address wording suggestions

NicolasHug commented on examples/inspection/plot_linear_model_coefficient_interpretation.py:
```diff
@@ -632,14 +647,25 @@
+#   between a given feature and the target, **conditional** on the other
+#   features.
+# * Correlated features induce instabilities in the coefficients of linear
+#   models and there effects cannot be well teased apart.
```
```suggestion
#   models and their effects cannot be well teased apart.
```

NicolasHug commented on examples/inspection/plot_linear_model_coefficient_interpretation.py:
```diff
@@ -632,14 +647,25 @@
 #
+# It is important to keep in mind that the coefficients that have been
+# dropped may still be related to the outcome by themselves: the model
+# chose to suppress them because they brings little or no additional
```
```suggestion
# chose to suppress them because they bring little or no additional
```

GaelVaroquaux commented:
@cmarmo  and @glemaitre I've addressed your comments. Thank you!

I think that it would be useful to have a review from a native-English speaker. @jnothman : you reviewed the original PR. This one should (hopefully) not take you long.

Thanks!

GaelVaroquaux committed:
Update examples/inspection/plot_linear_model_coefficient_interpretation.py

Co-Authored-By: Nicolas Hug <contact@nicolas-hug.com>

GaelVaroquaux committed:
Update examples/inspection/plot_linear_model_coefficient_interpretation.py

Co-Authored-By: Nicolas Hug <contact@nicolas-hug.com>

GaelVaroquaux commented:
To avoid straining our reviewing resources, if @jnothman doesn't have the time to look at this today, I suggest that we merge: reviewing time is a bottleneck, and Joel probably has better things to do than validating my nitpicks in English.

glemaitre commented:
Merging then with the current approvals.
